pkg/query/columnquery.go

```diff
@@ -732,9 +733,35 @@ func ComputeDiff(ctx context.Context, tracer trace.Tracer, base, compare profile
 	_, span := tracer.Start(ctx, "ComputeDiff")
 	defer span.End()
 
-	records := make([]arrow.Record, 0, len(compare.Samples)+len(base.Samples))
+	var (
+		totalCompareValue          int64
+		totalCompareValuePerSecond float64
+		totalBaseValue             int64
+		totalBaseValuePerSecond    float64
+	)
+	for _, s := range compare.Samples {
+		totalCompareValue += math.Int64.Sum(s.Column(len(s.Columns()) - 4).(*array.Int64))
+		totalCompareValuePerSecond += math.Float64.Sum(s.Column(len(s.Columns()) - 3).(*array.Float64))
+	}
+	for _, s := range base.Samples {
+		totalBaseValue += math.Int64.Sum(s.Column(len(s.Columns()) - 4).(*array.Int64))
+		totalBaseValuePerSecond += math.Float64.Sum(s.Column(len(s.Columns()) - 3).(*array.Float64))
+	}
 
+	records := make([]arrow.Record, 0, len(compare.Samples)+len(base.Samples))
 	for _, r := range compare.Samples {
+		var cumulativeRatio, cumulativePerSecondRatio float64
+		if totalBaseValue > totalCompareValue {
+			cumulativeRatio = float64(totalBaseValue) / float64(totalCompareValue)
```

Comment:
can we just move all these calculations before the for loops? then we'll just have a compareRatio, and a baseRatio that we multiply the respective sides by